utils/wui.py

```diff
@@ -0,0 +1,220 @@
+        # add footer with defines of files
+        self.finishMainFile()
+        # replace & remove
+        self.replaceOrigin()
+    #endef
+
+    # write data into a file
+    def writeTmpData(self, data):
+        tmp_file = open(self.raw_data_file_tmp, 'a')
+        tmp_file.writelines(data)
+        tmp_file.close()
+    #endef
+
+    # read file as binary and wrote hex data into a main file
+    def writeFile(self, file):
+        # lines to wrote into a tmp file
+        data = []
+        # file name
+        file_name = Path(file).name
+        
+        # define file array
+        data.append('static const unsigned char FSDATA_ALIGN_PRE data__' +file_name.replace('.', '_') + '[] FSDATA_ALIGN_POST = {\n')
+
+        # get hex of file name, add into text data array
+        file_name = '/' + file_name
+        hex_data = []
+        for ch in file_name:
+            ch_hex = '0x' + ch.encode('utf-8').hex()
+            hex_data.append(ch_hex)
+
+        # add null terminator
+        hex_data.append('0x00')
+        # generate string
+        hex_data = ','.join(hex_data)
+        hex_data += ',\n'
+        data.append(hex_data)
+
+        # hex file data 
+        #  hex_data = self.getHex(file)
+        hex_data = self.getHex(file + '.gz')
+        hex_data += ',\n'
+        data.append(hex_data)
+
+        # end of hex data array
+        data.append('};\n\n')
+
+        # write data
+        self.writeTmpData(data)
+    #endef
+    
+    # create new temporary file to write content into it then,
+    # after all content is made, this file will replace the original one
+    def createTmpFile(self):
+        open(self.raw_data_file_tmp, 'w')
+    #endef
+
+    def generateHeader(self):
+        # lines to wrote into a tmp file
+        data = []
+
+        data.append('#include "lwip/apps/fs.h"\n')
+        data.append('#include "lwip/def.h"\n\n')
+        
+        data.append('#define file_NULL (struct fsdata_file *)NULL\n\n')
+
+        data.append('#ifndef FS_FILE_FLAGS_HEADER_INCLUDED\n')
+        data.append('#define FS_FILE_FLAGS_HEADER_INCLUDED 1\n')
+        data.append('#endif\n')
+        data.append('#ifndef FS_FILE_FLAGS_HEADER_PERSISTENT\n')
+        data.append('#define FS_FILE_FLAGS_HEADER_PERSISTENT 0\n')
+        data.append('#endif\n')
+        data.append('#ifndef FSDATA_FILE_ALIGNMENT\n')
+        data.append('#define FSDATA_FILE_ALIGNMENT 0\n')
+        data.append('#endif\n')
+        data.append('#ifndef FSDATA_ALIGN_PRE\n')
+        data.append('#define FSDATA_ALIGN_PRE\n')
+        data.append('#endif\n')
+        data.append('#ifndef FSDATA_ALIGN_POST\n')
+        data.append('#define FSDATA_ALIGN_POST\n')
+        data.append('#endif\n')
+        data.append('#if FSDATA_FILE_ALIGNMENT == 2\n')
+        data.append('#include "fsdata_alignment.h"\n')
+        data.append('#endif\n\n')
+        #  data.append('')
+
+        # write data
+        self.writeTmpData(data)
+    #endef
+#enclass
```

Comment:
😳

Reply:
Sorry - old habits from projects I've done before. We used to mark end of each cycle, function, method, class, etc.. with these comments. I can delete them if you want.